src/cpp/fastdds/subscriber/DataReaderImpl.cpp

```diff
@@ -729,13 +729,18 @@ TypeSupport DataReaderImpl::type()
 }
 
 RequestedIncompatibleQosStatus& DataReaderImpl::update_requested_incompatible_qos(
-        QosPolicyId_t policy_id)
+        PolicyMask incompatible_policies)
 {
     ++requested_incompatible_qos_status_.total_count;
     ++requested_incompatible_qos_status_.total_count_change;
-    ++requested_incompatible_qos_status_.policies[policy_id].count;
-    requested_incompatible_qos_status_.last_policy_id = policy_id;
-
+    for (fastrtps::rtps::octet id = 0; id < NEXT_QOS_POLICY_ID; ++id)
+    {
+        if (incompatible_policies.test(id))
+        {
+            ++requested_incompatible_qos_status_.policies[static_cast<QosPolicyId_t>(id)].count;
+            requested_incompatible_qos_status_.last_policy_id = static_cast<QosPolicyId_t>(id);
+        }
+    }
```

Comment:
I think we should:
* Start the loop on the first policy, not on the invalid one
* Increase the total counts every time we increase the individual count

Reply:
same as above

Reply:
You're right.

Starting the loop on the first policy still applies, though.